scripts/installers/make-pyinstaller

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+"""Script to create a pyinstaller executable.
+
+This exe can then be wrapped in a platform specific installer for each
+supported platform.
+"""
+import argparse
+import os
+import sys
+import shutil
+import subprocess
+from distutils.dir_util import copy_tree
+
+
+ROOT = os.path.dirname(os.path.dirname(os.path.dirname(
+    os.path.abspath(__file__))))
+ASSET_DIR = os.path.join(ROOT, 'pyinstaller')
+EXE_NAME = 'aws'
+
+
+class BadRCError(Exception):
+    pass
+
+
+def run(cmd, cwd=None):
+    sys.stdout.write("Running cmd: %s\n" % cmd)
+    kwargs = {
+        'shell': True,
+        'stdout': subprocess.PIPE,
+        'stderr': subprocess.PIPE,
+    }
+    if cwd is not None:
+        kwargs['cwd'] = cwd
+    p = subprocess.Popen(cmd, **kwargs)
+    stdout, stderr = p.communicate()
+    rc = p.wait()
+    if p.returncode != 0:
+        raise BadRCError("Bad rc (%s) for cmd '%s': %s" % (
+            rc, cmd, stderr.decode() + stdout.decode()))
+    return stdout.decode() + stderr.decode()
+
+
+def make_pyinstaller(output_dir, cleanup=True):
+    delete_existing_exe(output_dir)
+    pyinstaller('aws.spec')
+    copy_dist_files(output_dir, 'aws', 'aws', create=True)
+    pyinstaller('aws_completer.spec')
+    copy_dist_files(output_dir, 'aws_completer', 'aws', create=False)
+    if cleanup:
+        cleanup_build()
+
+
+def delete_existing_exe(output_dir):
+    final_path = os.path.join(output_dir, EXE_NAME)
+    if os.path.isdir(final_path):
+        shutil.rmtree(final_path)
+    build_dir = os.path.join(ASSET_DIR, 'dist')
+    if os.path.isdir(build_dir):
+        shutil.rmtree(build_dir)
+
+
+def pyinstaller(specfile):
+    aws_spec_path = os.path.join(ASSET_DIR, specfile)
+    print(run('pyinstaller %s' % (aws_spec_path), cwd=ASSET_DIR))
+
+
+def copy_dist_files(output_dir, from_name, to_name, create):
+    src = os.path.join(ASSET_DIR, 'dist', from_name)
+    dst = os.path.join(output_dir, to_name)
+    print('maybe copying %s -> %s' % (src, dst))
+    if src != dst:
+        print('Copying %s -> %s' % (src, dst))
+        if create:
```

Comment:
Could you elaborate on why we need to use these two different functions and need the `create` parameter? I'd assume the logic would be check to make sure the output directory exists and if it does not exist then create the directory and finally do `shutil.copytree`.